app/notifications/rest.py

```diff
@@ -47,6 +54,12 @@ def create_sms_notification():
     except NoResultFound:
         return jsonify(result="error", message={'template': ['Template not found']}), 400
 
+    service = services_dao.dao_fetch_service_by_id(api_user['client'])
+
+    if service.restricted:
+        if notification['to'] not in [user.email_address for user in service.users]:
+            return jsonify(result="error", message={'to': ['Invalid phone number for restricted service']}), 400
+
```

Comment:
Thinking that the schema shouldn't do DB lookups? Was a bit worried about relatively hidden DB calls, wanted it all up front. Maybe not the best option, happy to refactor again when the service is on the context, as that will affect what calls we need to make.